lib/logstash/pipeline.rb

```diff
@@ -277,15 +259,12 @@ def shutdown
         # input teardown must be synchronized since is can be called concurrently by
         # the input worker thread and from the pipeline thread shutdown method.
         # this means that input teardown methods must support multiple calls.
-        @run_mutex.synchronize{input.teardown}
+        @run_mutex.synchronize{input.stop}
       rescue LogStash::ShutdownSignal
```

Comment:
is it still necessary anywhere? I was trying to find out why it was there in the first place, but couldn't find a reason.
I know it's supposed to isolate the "start input plugins" phase from a shutdown, but I couldn't see the need for the isolation.